src/System.Drawing.Common/src/System/Drawing/Pen.cs

```diff
@@ -558,6 +558,11 @@ public Color Color
             {
                 if (_color == Color.Empty)
                 {
+                    if (PenType != PenType.SolidColor)
+                    {
+                        throw new ArgumentException(SR.GdiplusInvalidParameter);
+                    }
```

Comment:
Maybe now that the behavior is the same in both GDI+ and libgdiplus we're anticipating the error in managed code instead of throwing it in `CheckStatus` https://github.com/dotnet/corefx/pull/29631/files/6b6aebfa300e8d5158c96822586e40b0bb21155a#diff-b0a2942015c2850f6dff6089cc5db47aR568 ?

However I think we should let the native libraries manage this errors, as this could change later, right?

Reply:
We need this now as older versions of libgdiplus do not return InvalidParameter. Although the compat bug has now been fixed, this retains compatability with older versions

Reply:
Makes sense.